### PR TITLE
6.0 & 6.x - Test Fix: Update test to properly reflect the Java Timestamp object.

### LINKED ISSUE #7843: 6.0 and 6.x build failure
The fix for 7719 causes a build failure on 6.x and 6.0.

The fix there changes from the Ruby Timestamp to the Java Timestamp, and the test needs to be updated to reflect that. 

```
Pending: (Failures listed here are expected and do not affect your suite's status)
  1) LogStash::Pipeline defaulting the pipeline workers based on thread safety when there are threadsafe filters only starts multiple filter threads
     # This test has been failing periodically since November 2016. Tracked as https://github.com/elastic/logstash/issues/6245
     # ./logstash-core/spec/logstash/pipeline_spec.rb:294
  2) LogStash::Agent#upgrade_pipeline when the upgrade fails and current state is empty should not start a pipeline
     # Temporarily skipped with xcontext
     # ./logstash-core/spec/logstash/agent_spec.rb:277
  3) LogStash::Modules::Scaffold import 4 realz puts stuff in ES
     # integration
     # ./logstash-core/spec/logstash/modules/scaffold_spec.rb:231
Failures:
  1) LogStash::Agent metrics after config reloading when reloading a good config sets the success reload timestamp
     Failure/Error: expect(value).to be_a(LogStash::Timestamp)
       expected #<Java::OrgLogstash::Timestamp:0x3f503565> to be a kind of LogStash::Timestamp
     # ./logstash-core/spec/logstash/agent_spec.rb:395:in `block in (root)'
     # ./vendor/bundle/jruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `block in /home/travis/build/elastic/logstash/vendor/bundle/jruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb'
     # ./lib/bootstrap/rspec.rb:13:in `<main>'
  2) LogStash::Agent metrics after config reloading when reloading a bad config sets the failure reload timestamp
     Failure/Error: expect(value).to be_a(LogStash::Timestamp)
       expected #<Java::OrgLogstash::Timestamp:0x4fbe9e80> to be a kind of LogStash::Timestamp
     # ./logstash-core/spec/logstash/agent_spec.rb:424:in `block in (root)'
     # ./vendor/bundle/jruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `block in /home/travis/build/elastic/logstash/vendor/bundle/jruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb'
     # ./lib/bootstrap/rspec.rb:13:in `<main>'
Finished in 8 minutes 2 seconds (files took 17.42 seconds to load)
2393 examples, 2 failures, 3 pending
Failed examples:
rspec ./logstash-core/spec/logstash/agent_spec.rb:392 # LogStash::Agent metrics after config reloading when reloading a good config sets the success reload timestamp
rspec ./logstash-core/spec/logstash/agent_spec.rb:421 # LogStash::Agent metrics after config reloading when reloading a bad config sets the failure reload timestamp
Randomized with seed 27715
```

### PR DESCRIPTION
**Pull request by jakelandis**
Fixes #7843 
This needs to be merged to 6.0 and 6.x, a fix for master will come on a different PR.

### CODE REVIEW

File: logstash-core/spec/logstash/agent_spec.rb [hunk 1]
```diff
@@ -9,6 +9,7 @@
 require_relative "../support/helpers"
 require_relative "../support/matchers"
 
+java_import org.logstash.Timestamp
```
Comment (original-brownbear) [inline logstash-core/spec/logstash/agent_spec.rb]: @jakelandis in the `master` version #7845 you have an empty line between this import and the `describe` line.
Maybe add that empty line here too, to keep the diff to `master` small, other than that also LGTM :)

Comment (jakelandis) [inline logstash-core/spec/logstash/agent_spec.rb]: @original-brownbear - thanks.  added the empty line to this PR.

Comment (original-brownbear) [inline logstash-core/spec/logstash/agent_spec.rb]: @jakelandis np + LGTM :)

**Comment by jakelandis** [comment]
Test failure not related: https://github.com/elastic/logstash/issues/7842

**Pull request closed without merging**